src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -383,19 +384,35 @@ private async Task<IActionResult> CreatePOST(string id, string returnUrl, bool s
 
             await conditionallyPublish(contentItem);
 
-            if ((!string.IsNullOrEmpty(returnUrl)) && (!stayOnSamePage))
+            if (createNew)
             {
-                return LocalRedirect(returnUrl);
-            }
+                var routeValueDictionnary = new RouteValueDictionary { { "id", id }, { "returnUrl", returnUrl } };
+                var layerMetadata = contentItem.As<LayerMetadata>();
 
```

Comment:
To be honest this would require a second thought. This makes the Contents module dependent on the Layers module which is not so good.